ui/dashboards/src/utils/panelUtils.ts

```diff
@@ -0,0 +1,96 @@
+// Copyright 2023 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { PanelDefinition } from '@perses-dev/core';
+import { PanelGroupDefinition } from '../context';
+
+export const removeWhiteSpacesAndSpecialCharacters = (str: string) => {
+  return str.replace(/\s+/g, '');
+};
+
+// Given a PanelGroup, will find the Y coordinate for adding a new row to the grid, taking into account the items present
+export function getYForNewRow(group: PanelGroupDefinition) {
```

Comment:
Helper extracted as is from `panel-editor-slice`, so it could also be used in `duplicate-panel-slice`.